File: Tutoriels/Cpp/OpenGL/z0_Ressources/src/manager/GProcess.cpp
//===============================================
#include "GProcess.h"
#include "GProcessWindow.h"
#include "GConfig.h"
//===============================================
GProcess::GProcess() {

}
//===============================================
GProcess::~GProcess() {

}
//===============================================
GProcess* GProcess::Instance() {
    string lProcess = GConfig::Instance()->getData("PROCESS");
    if(lProcess == "WINDOW") return GProcessWindow::Instance();
    return GProcessWindow::Instance();
}
//===============================================

File: Tutoriels/PHP/Projets/php/class/GProcess.php
<?php   
    abstract class GProcess {
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            $lKey = GConfig::Instance()->getData("PROCESS");
            if($lKey == "PRINT") return GProcessPrint::Instance();
            return GProcessPrint::Instance();
        }
        //===============================================
        abstract public function run();
        //===============================================
    }
?>